driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2282,6 +2285,195 @@ public void run() {
+                executor.submit(new ExceptionCatchingRunnable() {
+                    @Override
+                    public void runMayThrow() throws InterruptedException, ExecutionException {
+                        SchemaRefreshRequest coalesced = null;
+                        for (SchemaRefreshRequest request : events) {
+                            coalesced = coalesced == null ? request : coalesced.coalesce(request);
+                        }
+                        logger.trace("Coalesced schema refresh request: {}", coalesced);
+                        controlConnection.refreshSchema(coalesced.keyspace, coalesced.table);
+                    }
+                });
+            }
+
+        }
+
+        private class NodeRefreshRequest implements Delayed {
+
+            private final Host host;
+
+            private final HostEvent eventType;
+
+            private final long created;
+
+            private NodeRefreshRequest(Host host, HostEvent eventType) {
+                this.host = host;
+                this.eventType = eventType;
+                this.created = System.currentTimeMillis();
+            }
+
+            @Override
+            public long getDelay(TimeUnit unit) {
+                long now = System.currentTimeMillis();
+                long delayMillis = created + NEW_NODE_DELAY_SECONDS * 1000 - now;
+                return unit.convert(delayMillis, TimeUnit.MILLISECONDS);
+            }
+
+            @Override
+            public int compareTo(Delayed o) {
+                return ComparisonChain.start().compare(this.created, ((NodeRefreshRequest)o).created).result();
+            }
+
+            @Override
+            public String toString() {
+                return Objects.toStringHelper(this)
+                    .add("host", host)
+                    .add("eventType", eventType)
+                    .toString();
+            }
+
+        }
+
+        private class NodeRefreshRequestDeliveryCallback implements AbstractEventDebouncer.DeliveryCallback<NodeRefreshRequest> {
+
+            @Override
+            public void deliver(List<NodeRefreshRequest> events) {
+                Map<Host, HostEvent> hosts = new HashMap<Host, HostEvent>();
+                // only keep the last event for each host
+                for (NodeRefreshRequest req : events) {
+                    hosts.put(req.host, req.eventType);
+                }
+                for (final Entry<Host, HostEvent> entry : hosts.entrySet()) {
+                    final Host host = entry.getKey();
+                    HostEvent eventType = entry.getValue();
+                    // Make sure we  call controlConnection.refreshNodeInfo(host)
+                    // so that we have up-to-date infos on that host before adding it (so we typically
+                    // catch that an upgraded node uses a new cassandra version).
+                    switch (eventType) {
+                        case UP:
```

Comment:
UP and ADDED events are delayed by 1 second. Previously only ADDED events would be delayed.